CMake中将本项目作为CPP的静态库引用，在ld的时候会报错未定义的函数

尝试使用CMake添加外部项目来讲这个项目编译出静态库，使用如下的CMake，在ld到最后的时候会出现一个函数无法找到，尝试过交换c和cxx库的ld顺序，简单尝试了下，目前这个ld顺序的报错是最少的
install/strip和install以及开关 LTO都是一个报错，翻了下这个函数好像是在某个模板里面，但是不太了解CPP，不清楚应该如何修复这个问题
项目编译的cmake参数，参考build-aarch64-linux-gnu.sh脚本修改而来
如果作为动态库使用的话是没有问题
CMake:
```cmake
cmake_minimum_required(VERSION 3.19)
project(sherpa_onnx_report)

set(CMAKE_CXX_STANDARD 20)

include(ExternalProject)
set(DEPS_BUILD_DIR ${CMAKE_BINARY_DIR}/_deps)

find_program(CCACHE_FOUND ccache)
if(CCACHE_FOUND)
    message(STATUS "use ccache")
    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
endif(CCACHE_FOUND)

# --- ALSA (asound) ---
set(ALSA_SOURCE_DIR ${DEPS_BUILD_DIR}/alsa-src)
set(ALSA_BINARY_DIR ${DEPS_BUILD_DIR}/alsa-build)
ExternalProject_Add(ep_alsa
        GIT_REPOSITORY    https://github.com/alsa-project/alsa-lib.git
        GIT_TAG           v1.2.12
        GIT_SHALLOW       1
        CONFIGURE_COMMAND cd ${ALSA_SOURCE_DIR} && env "CC=${CMAKE_C_COMPILER}" "CXX=${CMAKE_CXX_COMPILER}" ./gitcompile --host=aarch64-linux-gnu
        BUILD_COMMAND     ""
        INSTALL_COMMAND   ""
        UPDATE_DISCONNECTED 1
        SOURCE_DIR        ${ALSA_SOURCE_DIR}
        BINARY_DIR        ${ALSA_BINARY_DIR}
)
add_library(alsa_asound_iface INTERFACE)
add_dependencies(alsa_asound_iface ep_alsa)
target_include_directories(alsa_asound_iface INTERFACE ${ALSA_SOURCE_DIR}/include)
target_link_directories(alsa_asound_iface INTERFACE ${ALSA_SOURCE_DIR}/src/.libs)
target_link_libraries(alsa_asound_iface INTERFACE asound)

# --- sherpa-onnx ---
set(SHERPA_ONNX_SOURCE_DIR ${DEPS_BUILD_DIR}/sherpa-onnx-src)
set(SHERPA_ONNX_BINARY_DIR ${DEPS_BUILD_DIR}/sherpa-onnx-build)
set(SHERPA_ONNX_INSTALL_DIR ${DEPS_BUILD_DIR}/sherpa-onnx-install)
ExternalProject_Add(ep_sherpa_onnx
        DEPENDS             ep_alsa
        GIT_REPOSITORY      https://github.com/k2-fsa/sherpa-onnx.git
        GIT_TAG             master
        GIT_SHALLOW         1
        UPDATE_DISCONNECTED 1
        SOURCE_DIR          ${SHERPA_ONNX_SOURCE_DIR}
        BINARY_DIR          ${SHERPA_ONNX_BINARY_DIR}
        INSTALL_DIR         ${SHERPA_ONNX_INSTALL_DIR}
        CONFIGURE_COMMAND   env CPLUS_INCLUDE_PATH=${ALSA_SOURCE_DIR}/include SHERPA_ONNX_ALSA_LIB_DIR=${ALSA_SOURCE_DIR}/src/.libs
                            cmake
                            -S <SOURCE_DIR>
                            -B <BINARY_DIR>
                            -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
                            -DCMAKE_BUILD_TYPE=Release
                            -DCMAKE_TOOLCHAIN_FILE=<SOURCE_DIR>/toolchains/aarch64-linux-gnu.toolchain.cmake
                            -DBUILD_SHARED_LIBS=OFF
                            -DSHERPA_ONNX_ENABLE_GPU=OFF -DBUILD_PIPER_PHONMIZE_EXE=OFF -DBUILD_PIPER_PHONMIZE_TESTS=OFF
                            -DBUILD_ESPEAK_NG_EXE=OFF -DBUILD_ESPEAK_NG_TESTS=OFF -DSHERPA_ONNX_ENABLE_TESTS=OFF
                            -DSHERPA_ONNX_ENABLE_PYTHON=OFF -DSHERPA_ONNX_ENABLE_CHECK=OFF -DSHERPA_ONNX_ENABLE_PORTAUDIO=OFF
                            -DSHERPA_ONNX_ENABLE_JNI=OFF -DSHERPA_ONNX_ENABLE_C_API=ON -DSHERPA_ONNX_ENABLE_WEBSOCKET=OFF
                            -DSHERPA_ONNX_LINK_LIBSTDCPP_STATICALLY=OFF -DSHERPA_ONNX_ENABLE_BINARY=OFF -DSHERPA_ONNX_ENABLE_TTS=OFF
                            -DSHERPA_ONNX_ENABLE_SPEAKER_DIARIZATION=OFF
        BUILD_COMMAND       make -j
        INSTALL_COMMAND     make install
)
add_library(sherpa_onnx_iface INTERFACE)
add_dependencies(sherpa_onnx_iface ep_sherpa_onnx)
target_include_directories(sherpa_onnx_iface INTERFACE ${SHERPA_ONNX_INSTALL_DIR}/include ${SHERPA_ONNX_INSTALL_DIR}/include/sherpa-onnx)
target_link_directories(sherpa_onnx_iface INTERFACE ${SHERPA_ONNX_INSTALL_DIR}/lib)
target_link_libraries(sherpa_onnx_iface INTERFACE
        sherpa-onnx-cxx-api
        sherpa-onnx-c-api
        sherpa-onnx-core
        kaldi-decoder-core
        sherpa-onnx-kaldifst-core
        sherpa-onnx-fst
        kaldi-native-fbank-core
        kissfft-float
        onnxruntime
        ssentencepiece_core
)

include(CheckIPOSupported)
check_ipo_supported(RESULT LTO_IS_SUPPORTED)
if(LTO_IS_SUPPORTED)
    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
    message(STATUS "LTO/IPO enabled for the App project.")
else()
    message(WARNING "LTO/IPO is not supported by the toolchain: ${message}")
endif()

add_executable(sherpa_onnx_report main.cpp)

add_dependencies(sherpa_onnx_report
        ep_alsa
        ep_sherpa_onnx
)

target_link_libraries(sherpa_onnx_report PRIVATE
        alsa_asound_iface
        sherpa_onnx_iface
)

```

Toolchain CMake:
```cmake
# Copied from https://github.com/Tencent/ncnn/blob/master/toolchains/aarch64-linux-gnu.toolchain.cmake
set(CMAKE_SYSTEM_NAME Linux)
set(CMAKE_SYSTEM_PROCESSOR aarch64)

set(CMAKE_C_COMPILER_TARGET "aarch64-linux-gnu")
set(CMAKE_C_COMPILER "${CMAKE_C_COMPILER_TARGET}-gcc")
set(CMAKE_CXX_COMPILER "${CMAKE_C_COMPILER_TARGET}-g++")

#if(NOT CMAKE_FIND_ROOT_PATH_MODE_PROGRAM)
#    set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
#endif()
#if(NOT CMAKE_FIND_ROOT_PATH_MODE_LIBRARY)
#    set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
#endif()
#if(NOT CMAKE_FIND_ROOT_PATH_MODE_INCLUDE)
#    set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
#endif()
#if(NOT CMAKE_FIND_ROOT_PATH_MODE_PACKAGE)
#    set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
#endif()

set(CMAKE_C_FLAGS "-march=armv8-a")
set(CMAKE_CXX_FLAGS "-march=armv8-a")

# cache flags
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}" CACHE STRING "c flags")
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" CACHE STRING "c++ flags")
```

CMake Log:
```bash
/home/fan/.cache/JetBrains/RemoteDev/dist/42b01a56714d5_CLion-2025.1.2/bin/cmake/linux/x64/bin/cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_MAKE_PROGRAM=/home/fan/.cache/JetBrains/RemoteDev/dist/42b01a56714d5_CLion-2025.1.2/bin/ninja/linux/x64/ninja -DCMAKE_TOOLCHAIN_FILE=toolchain_aarch64.cmake -G Ninja -S /home/fan/CLionProjects/sherpa-onnx-report -B /home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/aarch64-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/aarch64-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- use ccache
-- LTO/IPO enabled for the App project.
-- Configuring done (1.4s)
-- Generating done (0.0s)
-- Build files have been written to: /home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo
```
编译 Log:
```bash
[18/18] Linking CXX executable sherpa_onnx_report
FAILED: sherpa_onnx_report 
: && ccache /usr/bin/aarch64-linux-gnu-g++ -march=armv8-a -O2 -g -DNDEBUG -flto=auto -fno-fat-lto-objects -Wl,--dependency-file=CMakeFiles/sherpa_onnx_report.dir/link.d CMakeFiles/sherpa_onnx_report.dir/main.cpp.o -o sherpa_onnx_report -L/home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo/_deps/alsa-src/src/.libs   -L/home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo/_deps/sherpa-onnx-install/lib -Wl,-rpath,/home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo/_deps/alsa-src/src/.libs:/home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo/_deps/sherpa-onnx-install/lib  -lasound  -lsherpa-onnx-cxx-api  -lsherpa-onnx-c-api  -lsherpa-onnx-core  -lkaldi-decoder-core  -lsherpa-onnx-kaldifst-core  -lsherpa-onnx-fst  -lkaldi-native-fbank-core  -lkissfft-float  -lonnxruntime  -lssentencepiece_core && :
/usr/lib/gcc-cross/aarch64-linux-gnu/13/../../../../aarch64-linux-gnu/bin/ld: /home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo/_deps/sherpa-onnx-install/lib/libsherpa-onnx-core.a(offline-recognizer-impl.cc.o): in function `fst::FarReader<fst::ArcTpl<fst::TropicalWeightTpl<float> > >::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
offline-recognizer-impl.cc:(.text._ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0xc0): undefined reference to `fst::IsSTTable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/lib/gcc-cross/aarch64-linux-gnu/13/../../../../aarch64-linux-gnu/bin/ld: offline-recognizer-impl.cc:(.text._ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x12c): undefined reference to `fst::IsSTList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
完整日志:

[full.log](https://github.com/user-attachments/files/20959361/full.log)

> 1. 请问你有无修改 sherpa-onnx? 如果有，请别修改，然后重试。
> 2. 请问你是否用的最新的 sherpa-onnx master? 如果是，请贴 log, 证明你用的是 master. 如果不是，请用 master
> 3. 请问你的 库，和 app, 是否是在同一台电脑，用同一个 版本的 gcc 编译的？如果不是，请重试，用一样的.
> 
> 如果 1, 2, 3都没问题,
> 
> ```
> (.text._ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0xc0): undefined reference to `fst::IsSTTable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
> /usr/lib/gcc-cross/aarch64-linux-gnu/13/../../../../aarch64-linux-gnu/bin/ld: offline-recognizer-impl.cc:(.text._ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN3fst9FarReaderINS_6ArcTplINS_17TropicalWeightTplIfEEEEE4OpenERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x12c): undefined reference to `fst::IsSTList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
> collect2: error: ld returned 1 exit status
> ```
> 
> 在链接的 库里面， 用 `nm` 去找一下，这个报错的符号，在哪个库里，把你跑的命令截图，证明你真的有去找了。

1 没有修改
2 是master
库是在cmake里面作为外部项目添加的，所以1，2的问题肯定是不会有的，cmake会从github去clone master分支下来
```cmake
ExternalProject_Add(ep_sherpa_onnx
        DEPENDS             ep_alsa
        GIT_REPOSITORY      https://github.com/k2-fsa/sherpa-onnx.git
        GIT_TAG             master
```
我项目的cmake配置工具链输出
```bash
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/aarch64-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/aarch64-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- use ccache
-- Configuring done (0.9s)
-- Generating done (0.0s)
-- Build files have been written to: /home/fan/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo
```
库的cmake配置工具链输出
```bash
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/aarch64-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/aarch64-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
```
3 库和app都在同一个Ubuntu noble的X86主机上使用同一个arm64工具链交叉编译的，使用的都是系统源安装的交叉工具链，只安装了一个gcc 13的工具链

nm查看的符号
```bash
fan@ub24:~/CLionProjects/sherpa-onnx-report/cmake-build-relwithdebinfo/_deps/sherpa-onnx-install/lib$ nm -A --demangle *.a | grep -E "fst::IsSTTable|fst::IsSTList"
libsherpa-onnx-core.a:offline-recognizer-impl.cc.o:                 U fst::IsSTList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
libsherpa-onnx-core.a:offline-recognizer-impl.cc.o:                 U fst::IsSTTable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
libsherpa-onnx-core.a:online-recognizer-impl.cc.o:                 U fst::IsSTList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
libsherpa-onnx-core.a:online-recognizer-impl.cc.o:                 U fst::IsSTTable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
libsherpa-onnx-fstfar.a:sttable.cc.o:0000000000000000 T fst::IsSTTable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
libsherpa-onnx-fstfar.a:stlist.cc.o:0000000000000000 T fst::IsSTList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
```

--- a/.github/workflows/pkg-config.yaml
+++ b/.github/workflows/pkg-config.yaml
@@ -33,13 +33,14 @@ concurrency:
 jobs:
   pkg_config:
     runs-on: ${{ matrix.os }}
-    name: ${{ matrix.os }} ${{ matrix.build_type }} ${{ matrix.lib_type }}
+    name: ${{ matrix.os }} ${{ matrix.build_type }} ${{ matrix.lib_type }} tts-${{ matrix.tts }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
         build_type: [Release, Debug]
         lib_type: [shared, static]
+        tts: [ON, OFF]
 
     steps:
       - uses: actions/checkout@v4
@@ -67,12 +68,12 @@ jobs:
           mkdir build
           cd build
           if [[ ${{ matrix.lib_type }} == "shared" ]]; then
-            cmake -DSHERPA_ONNX_ENABLE_C_API=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_INSTALL_PREFIX=./install -DBUILD_SHARED_LIBS=ON ..
+            cmake -DSHERPA_ONNX_ENABLE_TTS=${{ matrix.tts }} -DSHERPA_ONNX_ENABLE_C_API=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_INSTALL_PREFIX=./install -DBUILD_SHARED_LIBS=ON ..
           else
-            cmake -DSHERPA_ONNX_ENABLE_C_API=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_INSTALL_PREFIX=./install -DBUILD_SHARED_LIBS=OFF ..
+            cmake -DSHERPA_ONNX_ENABLE_TTS=${{ matrix.tts }} -DSHERPA_ONNX_ENABLE_C_API=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_INSTALL_PREFIX=./install -DBUILD_SHARED_LIBS=OFF ..
           fi
 
-      - name: Build sherpa-onnx for ${{ matrix.os }} ${{ matrix.build_type }} ${{ matrix.lib_type }}
+      - name: Build sherpa-onnx for ${{ matrix.os }} ${{ matrix.build_type }} ${{ matrix.lib_type }} tts-${{ matrix.tts }}
         shell: bash
         run: |
           export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
@@ -104,16 +105,36 @@ jobs:
 
           cat build/install/sherpa-onnx.pc
 
+      - name: Show pkg-config
+        shell: bash
+        run: |
+          export PKG_CONFIG_PATH=$PWD/build/install:$PKG_CONFIG_PATH
+          pkg-config --cflags sherpa-onnx
+          pkg-config --libs sherpa-onnx
+
       - name: Build C API example
         shell: bash
         run: |
           export PKG_CONFIG_PATH=$PWD/build/install:$PKG_CONFIG_PATH
           cd c-api-examples
 
+          pkg-config --cflags sherpa-onnx
+
           gcc -o decode-file-c-api $(pkg-config --cflags sherpa-onnx) ./decode-file-c-api.c $(pkg-config --libs sherpa-onnx)
+
           ./decode-file-c-api --help
 
+      - name: Build C API example (tts)
+        if: matrix.tts == 'ON'
+        shell: bash
+        run: |
+          export PKG_CONFIG_PATH=$PWD/build/install:$PKG_CONFIG_PATH
+          cd c-api-examples
+
+          pkg-config --cflags sherpa-onnx
+
           gcc -o offline-tts-c-api $(pkg-config --cflags sherpa-onnx) ./offline-tts-c-api.c $(pkg-config --libs sherpa-onnx)
+
           ./offline-tts-c-api --help
 
       - name: Test online transducer (C API)
@@ -126,5 +147,5 @@ jobs:
 
       - uses: actions/upload-artifact@v4
         with:
-          name: tts-generated-test-files-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.lib_type }}
+          name: tts-generated-test-files-${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.lib_type }}-tts-${{ matrix.tts }}
           path: tts

--- a/cmake/sherpa-onnx-static-no-tts.pc.in
+++ b/cmake/sherpa-onnx-static-no-tts.pc.in
@@ -22,4 +22,4 @@ Cflags: -I"${includedir}"
 # Note: -lcargs is required only for the following file
 # https://github.com/k2-fsa/sherpa-onnx/blob/master/c-api-examples/decode-file-c-api.c
 # We add it here so that users don't need to specify -lcargs when compiling decode-file-c-api.c
-Libs: -L"${libdir}" -lsherpa-onnx-c-api -lsherpa-onnx-core -lkaldi-decoder-core -lsherpa-onnx-kaldifst-core -lsherpa-onnx-fstfar  -lsherpa-onnx-fst -lkaldi-native-fbank-core -lkissfft-float -lonnxruntime -lssentencepiece_core -Wl,-rpath,${libdir} @SHERPA_ONNX_PKG_WITH_CARGS@ @SHERPA_ONNX_PKG_CONFIG_EXTRA_LIBS@
+Libs: -L"${libdir}" -lsherpa-onnx-c-api -lsherpa-onnx-core -lkaldi-decoder-core -lsherpa-onnx-kaldifst-core -lsherpa-onnx-fstfar -lsherpa-onnx-fst -lkaldi-native-fbank-core -lkissfft-float -lonnxruntime -lssentencepiece_core -Wl,-rpath,${libdir} @SHERPA_ONNX_PKG_WITH_CARGS@ @SHERPA_ONNX_PKG_CONFIG_EXTRA_LIBS@

--- a/cmake/sherpa-onnx-static-no-tts.pc.in
+++ b/cmake/sherpa-onnx-static-no-tts.pc.in
@@ -22,4 +22,4 @@ Cflags: -I"${includedir}"
 # Note: -lcargs is required only for the following file
 # https://github.com/k2-fsa/sherpa-onnx/blob/master/c-api-examples/decode-file-c-api.c
 # We add it here so that users don't need to specify -lcargs when compiling decode-file-c-api.c
-Libs: -L"${libdir}" -lsherpa-onnx-c-api -lsherpa-onnx-core -lkaldi-decoder-core -lsherpa-onnx-kaldifst-core -lsherpa-onnx-fst -lkaldi-native-fbank-core -lkissfft-float -lonnxruntime -lssentencepiece_core -Wl,-rpath,${libdir} @SHERPA_ONNX_PKG_WITH_CARGS@ @SHERPA_ONNX_PKG_CONFIG_EXTRA_LIBS@
+Libs: -L"${libdir}" -lsherpa-onnx-c-api -lsherpa-onnx-core -lkaldi-decoder-core -lsherpa-onnx-kaldifst-core -lsherpa-onnx-fstfar  -lsherpa-onnx-fst -lkaldi-native-fbank-core -lkissfft-float -lonnxruntime -lssentencepiece_core -Wl,-rpath,${libdir} @SHERPA_ONNX_PKG_WITH_CARGS@ @SHERPA_ONNX_PKG_CONFIG_EXTRA_LIBS@

